src/apiTest/kotlin/com/ampnet/auditornode/ApiTestWithPropertiesBase.kt

```diff
@@ -0,0 +1,56 @@
+package com.ampnet.auditornode
+
+import io.micronaut.context.env.PropertySource
+import io.micronaut.http.client.RxHttpClient
+import io.micronaut.runtime.server.EmbeddedServer
+import io.micronaut.test.extensions.AbstractMicronautExtension
+import io.micronaut.test.support.server.TestExecutableEmbeddedServer
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.TestInstance
+import javax.inject.Inject
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+abstract class ApiTestWithPropertiesBase : TestBase() {
+
+    @Inject
+    // request path must always include serverPath(), e.g. "${serverPath()}/hello"
+    protected lateinit var client: RxHttpClient
+
+    @Inject
+    private lateinit var server: EmbeddedServer
+
+    abstract val propertySourceName: String
+
+    @BeforeAll
+    fun beforeAll() {
+        // This logic is necessary when running tests against the native image - for some reason, Micronaut wants to use
+        // only the property source named "test-properties" (AbstractMicronautExtension.TEST_PROPERTY_SOURCE) so this
+        // will rename the specified property source to match the name that Micronaut wants.
+        if (System.getProperty(TestExecutableEmbeddedServer.PROPERTY) != null) {
+            if (server.isRunning) {
+                server.stop()
+            }
+
+            val scriptPropertySource = server.environment.propertySources.find { it.name == propertySourceName }!!
```

Comment:
```suggestion
            val scriptPropertySource = server.environment.propertySources.find { it.name == propertySourceName } ?: fail("Missing property source")
```
Try to use `fail`, this provides clear code and an error will be provided with the message.